packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,281 @@
+  margin: 0;
+`;
+
+const EscapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies that the Modal content will appear portaled to the end of the DOM,
+   * rather than in the React DOM tree.
+   *
+   * default: `true`
+   */
+  usePortal?: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `normal`
+   */
+  size?: ModalSize;
+
+  /**
+   * Specifies the title of the Modal.
+   *
+   */
+  title?: string;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback invoked when Modal closes.
+   *
+   */
+  onRequestClose?: () => void;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  usePortal
+  size="large"
+  setActive={setActive}
+  title="My Modal"
+  onRequestClose={() => console.log('Modal is closing now!')}
+  >  
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.usePortal Boolean to describe if content should be portaled to end of DOM, or appear in React DOM tree.
+ * @param props.size String to determine size of Modal. ['xxsmall', 'xsmall', 'small', 'normal', 'large', 'xlarge']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.title Title for the Modal, will appear inside header tags.
+ * @param props.onRequestClose Callback invoked when Modal is closed.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  usePortal = true,
+  size = ModalSize.Normal,
+  setActive,
+  children,
+  title,
+  onRequestClose,
+  className,
+  ...rest
+}: ModalProps) {
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = () => {
+    if (onRequestClose) {
+      onRequestClose();
+    }
+
+    if (setActive) {
+      setActive(false);
+    }
+  };
+
+  const handleEscape = useCallback((e: KeyboardEvent) => {
+    if (e.keyCode === EscapeKey) {
+      handleClose();
+    }
+  }, []);
+
+  const handleDocumentClick = (e: React.SyntheticEvent) => {
+    if (!contentRef.current) {
+      return;
+    }
+
+    if (!contentRef.current.contains(e.target as Node)) {
+      handleClose();
+    }
+  };
+
+  useEventListener('keydown', handleEscape, {
+    options: { once: true },
+  });
+
+  const Root = usePortal ? Portal : Fragment;
+
+  return active ? (
+    <Root>
+      <div {...rest} className={overlayStyle} onClick={handleDocumentClick}>
+        <div
```

Comment:
If we make the modal and the overlay siblings instead of nesting the modal content, we can eliminate the `contains` check in `handleDocumentClick`, and probably remove the event handler from this element.

Reply:
Also we could remove all of the flex rules for the overlay!

Reply:
👍